flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/datagenerator/common/LabeledPointWithWeightGenerator.java

```diff
@@ -0,0 +1,194 @@
+ */
+
+package org.apache.flink.ml.benchmark.datagenerator.common;
+
+import org.apache.flink.api.common.functions.RichMapFunction;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.ml.benchmark.datagenerator.InputDataGenerator;
+import org.apache.flink.ml.benchmark.datagenerator.param.HasVectorDim;
+import org.apache.flink.ml.common.datastream.TableUtils;
+import org.apache.flink.ml.common.param.HasFeaturesCol;
+import org.apache.flink.ml.common.param.HasLabelCol;
+import org.apache.flink.ml.common.param.HasWeightCol;
+import org.apache.flink.ml.linalg.Vectors;
+import org.apache.flink.ml.linalg.typeinfo.DenseVectorTypeInfo;
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.util.ParamUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.NumberSequenceIterator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+/** A DataGenerator which creates a table of features, label and weight. */
+public class LabeledPointWithWeightGenerator
+        implements InputDataGenerator<LabeledPointWithWeightGenerator>,
+                HasFeaturesCol<LabeledPointWithWeightGenerator>,
+                HasLabelCol<LabeledPointWithWeightGenerator>,
+                HasWeightCol<LabeledPointWithWeightGenerator>,
+                HasVectorDim<LabeledPointWithWeightGenerator> {
+
+    public static final Param<Integer> FEATURE_ARITY =
+            new IntParam(
+                    "featureArity",
+                    "Arity of each feature. "
+                            + "If set to positive value, each feature would be an integer in range [0, arity - 1]. "
+                            + "If set to zero, each feature would be a continuous double in range [0, 1).",
+                    2,
+                    ParamValidators.gtEq(0));
+
+    public static final Param<Integer> LABEL_ARITY =
+            new IntParam(
+                    "labelArity",
+                    "Arity of label. "
+                            + "If set to positive value, the label would be an integer in range [0, arity - 1]. "
+                            + "If set to zero, the label would be a continuous double in range [0, 1).",
+                    2,
+                    ParamValidators.gtEq(0));
+
+    public static final Param<Integer> WEIGHT_ARITY =
```

Comment:
For benchmark, should we always generate weight as a random value in range [0, 1)? If not, what would be the other use-cases? If yes, maybe we can remove this parameter?

Reply:
I agree. I'll make the change.